controllers/profiles/followAnonUser.js

```diff
@@ -0,0 +1,105 @@
+const moment = require('moment');
+
+const UserFollowUserFunction = require('../../databases/functions/userFollowUser');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+const SuccessResponse = require('../../utils/response/SuccessResponse');
+
+const {UserFollowUser, UserFollowUserHistory, User, sequelize} = require('../../databases/models');
+const Getstream = require('../../vendor/getstream');
+const {addForFollowUser} = require('../../services/score');
+const BetterSocialCore = require('../../services/bettersocial');
+const UsersFunction = require('../../databases/functions/users');
+const {sendFollowMainFeedF2} = require('../../services/queue/mainFeedF2');
+
+module.exports = async (req, res) => {
+  const {user_id_followed, follow_source} = req.body;
+  const {user} = req;
+
+  if (req?.userId === user_id_followed)
+    return ErrorResponse.e403(res, 'Only allow following other profiles');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code checks if the user is trying to follow themselves and returns an error if true. This is a good check to prevent users from following themselves.

```diff
  if (req?.userId === user_id_followed)
    return ErrorResponse.e403(res, 'Only allow following other profiles');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->